gulp/tasks/build.js

```diff
@@ -182,13 +169,24 @@ function compileIndvES2015ModulesToBrowser() {
         }
       }),
       new webpack.optimize.UglifyJsPlugin({
-        sourceMap: true
+        sourceMap: true,
+        mangle: {
+          props: {
+            ignore_quoted: true,
+            regex: /_$/,
```

Comment:
Nice find on this, I found that by expanding this regex I was able to get some other small savings. The regex I used included the `_` prefix:

_i.e._
```javascript
/_$|^_/
```

I think, once this is in, going through and refactoring property names that don't get exposed (regardless of `public` or `private` markings as I know we use some of those for internally public stuff) we can get some more savings. 

### Size Comparison

| File | Suffix Only Regex | Prefix + Suffix Regex |
| --- | --- | --- |
| firebase-app.js | 16.14 KB (5.61 KB) | 15.7 KB (5.55 KB) |
| firebase-database.js | 166.16 KB (45.55 KB) | 165.8 KB (45.52 KB) |
| firebase-messaging.js | 18.22 KB (5.59 KB) | 18.2 KB (5.59 KB) |
| firebase.js | 371.3 KB (108.1 KB) | 370.49 KB (108.02 KB) |